portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -14,53 +14,55 @@
 
 package com.google.sps.data;
 
-import java.util.Date;
-import java.text.DateFormat;
 import java.lang.Comparable;
 import com.google.appengine.api.datastore.Entity;
-import javax.servlet.http.HttpServletRequest;
+import com.google.appengine.api.datastore.Key;
 
 /** Represents a single comment from a user. */
 public class Comment implements Comparable<Comment> {
-  public static final String COMMENT_NAME = "name";
-  public static final String COMMENT_TEXT = "comment";
-  public static final String COMMENT_TIMESTAMP = "timestamp";
+  /** Key for the name of the commenter */
+  static final String COMMENT_NAME = "name";
+  /** Key for the comment contents */
+  static final String COMMENT_TEXT = "comment";
+  /** Key for the timestamp */
+  static final String COMMENT_TIMESTAMP = "timestamp";
 
+  /** Internal counter used for dynamically assigning an ID */
+  private static int commentCount = 0;
+
+  /** Text of the comment */
   private final String text;
+  /** Name of the person who submitted the comment */
   private final String name;
+  /** Time comment was created, in milliseconds since epoch (UTC) */
   private final long timestamp;
+  /** Temporary ID used on the front-end */
+  private final int id;
+  /** The key associated with the entity that keeps this comment persist in storage. */
+  private final Key key;
 
-  private Comment(String text, String name) {
-    this(text, name, System.currentTimeMillis());
-  }
-
-  /** Create a comment with a posted date of given milliseconds from the epoch */
-  private Comment(String text, String name, long timestamp) {
+  /** Create a comment with a posted date of given milliseconds from the epoch. */
+  public Comment(String text, String name, Key key, long timestamp) {
     this.text = text;
     this.name = name;
     this.timestamp = timestamp;
+    this.key = key;
+
+    // Dynamically assign ID
+    id = commentCount++;
```

Comment:
I think that this implementation has a subtle bug, because it relies upon commentCount, different java servers run this program will have different values, sometimes colliding on values and sometimes not. 

you can experience this by creating a few comments, then restarting your server. I would imagine that the data in the database would look like this:

comment 0, comment 1, comment 2 (restart) comment 0

Reply:
this process may lead you to no longer use the id pattern that you have. .

Reply:
It never had comment ID's colliding, but for all the wrong reasons -- ID was never stored, so comments would have a different ID each time the server restarted. 

Needless to say, @yutaow's suggestion makes this much cleaner by just using `key` as an ID.